src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -1581,12 +1641,15 @@ public int doFinal(ByteBuffer src, ByteBuffer dst)
             if (mismatch != 0) {
                 // Clear output data
                 dst.reset();
-                if (dst.hasArray()) {
-                    int ofs = dst.arrayOffset() + dst.position();
-                    Arrays.fill(dst.array(), ofs , ofs + len, (byte)0);
-                } else {
-                    Unsafe.getUnsafe().setMemory(((DirectBuffer)dst).address(),
-                        len + dst.position(), (byte)0);
+                // If this is an in-place array, don't zero the src
```

Comment:
The comment doesn't jive with the line of code on the next line. It is the inverse of the comment.

Reply:
ok